packages/runtime/src/store/tutorial-runner.ts

```diff
@@ -189,6 +189,62 @@ export class TutorialRunner {
     );
   }
 
+  async fileExists(filepath: string) {
+    const previousLoadPromise = this._currentLoadTask?.promise;
+
+    return new Promise<boolean>((resolve, reject) => {
+      this._currentLoadTask = newTask(
+        async (signal) => {
+          await previousLoadPromise;
+
+          const webcontainer = await this._webcontainer;
+
+          if (signal.aborted) {
+            reject(new Error('Task was aborted'));
+          }
+
+          signal.throwIfAborted();
+
+          try {
+            await webcontainer.fs.readFile(filepath);
+            resolve(true);
+          } catch {
+            resolve(false);
+          }
+        },
+        { ignoreCancel: true },
+      );
+    });
+  }
+
+  async folderExists(folderPath: string) {
```

Comment:
Oh maybe we could create a helper function that both `folderExists` and `fileExists` call? Wdyt?